doc/man5/x509v3_config.pod

```diff
@@ -197,14 +197,17 @@ indicated by putting a colon C<:> between the value and this option.
 By default the B<x509>, B<req>, and B<ca> apps behave as if
 "none" was given for self-signed certificates and "keyid, issuer" otherwise.
 
-If B<keyid> is present, an attempt is made to compute the hash of the public key
-corresponding to the signing key in case the certificate is self-signed,
-or else to copy the subject key identifier (SKID) from the issuer certificate.
+If B<keyid> is present, an attempt is made to
+copy the subject key identifier (SKID) from the issuer certificate
+(which is the same as the current certificate if it is self-issued), where
```

Comment:
```suggestion
(which is the same as the current certificate if it is self-signed), where
```

Reply:
I deliberately used "self-issued" at this point because it is more correct/general:
The point here is issuance (which means who signed the certificate).
Note that the cert may still not be self-signed in this case because a different pubkey may be placed in it, using `-force_pubkey`.

Reply:
... except that *in code*, you check with `cert_matches_key()` everywhere, which is effectively a check to see if the cert is self-*signed*, not self-issued.

Or am I missing something?

Reply:
Things are a bit tricky, and it turns out that I had to further adapt both the doc and the code.
The check using `cert_matches_key()` prevents inclusion of a wrong fallback AKID for self-signed certs - this is fine here, but it turns out that for default AKID production `X509V3_set_issuer_pkey()` sill need to be called later.
For certs that are self-issued but not self-signed, the issuer cert may be given as a different cert with the same subject using the `-CA` option. This was so far not covered by the above doc line; I changed it to:
```
(which may be the same as the current certificate if it is self-issued), where
```
For the case explicitly mentioned there that we wanna produce an self-issued cert and the issuer cert is the same as the cert to be produced, the self-signed case was coded fine,
but when using `-force_pubkey` the SKID produced by default is wrong.
I plan to fix this further bug later today...

Reply:
> ... except that _in code_, you check with `cert_matches_key()` everywhere, which is effectively a check to see if the cert is self-_signed_, not self-issued.
> 
> Or am I missing something?

You're not missing anything.  The documentation in question is trying to distinguish between certificates where the issuer key and name are in that same cert (it is self-signed) and other certificates (possibly self-issued) where the issuer key comes from a separate issuer certificate (which might share the same DN, but has a different key).  See RFC5280 Section 3.2:
```
   Self-signed certificates are self-issued certificates where the digital
   signature may be verified by the public key bound into the certificate.
```

Reply:
> I deliberately chose "self-issued" at this point because it is more correct/general:
> The point here is issuance (which means who signed the certificate).
> Note that a self-issued cert may still not be self-signed in this case because a different pubkey may be placed in it, e.g., using `-force_pubkey`.

That choice is wrong.  It is not the appropriate distinction.

Reply:
> > I deliberately chose "self-issued" at this point because it is more correct/general:
> > The point here is issuance (which means who signed the certificate).
> > Note that a self-issued cert may still not be self-signed in this case because a different pubkey may be placed in it, e.g., using `-force_pubkey`.
> 
> That choice is wrong. It is not the appropriate distinction.

Why do you think so?
How else would you address the case that a self-issued but not self-signed cert is produced?

Reply:
Self-issued certificates have an issuer CA cert that happens to have the same subject DN, but a different key and SKID.  AKID `keyid` handling is the same for cross certs and self-issued certs, the AKID keyid is a copy of the issuer's SKID.  The special case is self-signed, not self-issued.

Reply:
> Self-issued certificates have an issuer CA cert that happens to have the same subject DN, but a different key and SKID.

Well, not quite: also self-signed certs (having same key and SKID) are self-issued,
while my question was how to cover in the above documentation sentence the case
of those self-issued certs that are not self-signed.

> AKID `keyid` handling is the same for cross certs and self-issued certs, the AKID keyid is a copy of the issuer's SKID. The special case is self-signed, not self-issued.

Good point that conceptually the special case for AKID production are self-signed certs,
simply because their AKID is typically omitted since it is the same as the SKID.

Reply:
> AKID `keyid` handling is the same for cross certs and self-issued certs, the AKID keyid is a copy of the issuer's SKID. The special case is self-signed, not self-issued.

It turns out there is a further special case:
the case that the issuer cert is the same as the subject cert and is self-issued but not self-signed.
This is now fixed both in the code and the doc.